joomla/joomla-cms
# [3.9] Align PHP namespaced Text with the javascript JText

Pull Request for Issue https://github.com/joomla/joomla-cms/pull/18912#discussion_r153995901 .

### Summary of Changes
- Normalise the method name for both PHP and JS world!
- Add appropriate deprecation
- Proxy Joomla.JText.* to Joomla.Text.*

### Testing Instructions
Apply PR
Got to Admin -> Global configuration
Open the browser's console
Copy paste:
- `Joomla.JText._('ERROR')` press enter you should see `"Error"`
- `Joomla.Text._('ERROR')` press enter you should see `"Error"`
- `Joomla.JText.strings` press enter you should see an object with all the available strings
- `Joomla.Text.strings` press enter you should see the same object with all the available strings as before
- `Joomla.Text.load({TEST: "test"})` and press enter then `Joomla.Text.strings` and check that TEST is there
- `Joomla.Text.load({TESTA: "testa"})` and press enter then `Joomla.Text.strings` and check that TESTA is there

### Expected result
Everything works as usual

### Actual result
<img width="1278" alt="screen shot 2017-12-31 at 00 35 46" src="https://user-images.githubusercontent.com/3889375/34458001-04c7fa80-edc3-11e7-969c-e5729979fb75.png">

### Documentation Changes Required
There are a bunch of console warnings added to help 3RD PD move forward. There will be a section in the documentation that will cover all the changes for J4.0

@mbabker up to you...
